### DISCUSSION TITLE
How to generate images with below various effects?

### DISCUSSION THREAD

**Discussion by nissansz** [python-pillow/Pillow#7863] [General]
![image](https://github.com/python-pillow/Pillow/assets/41010669/b1413e90-f688-4a5f-90de-9afa2b0c28d1)

**Comment by radarhere** [comment]
Hi. 'various effects' is very vague, and even more so when you are referring what looks like 14 different images.

Could you be more specific?

**Comment by nissansz** [comment]
Yes, I need similar effects in above image and more effects if available.

**Reply by radarhere** [reply]
It is not clear what effects in the images you are referring to.

Regarding 'more effects', Pillow is an image processing library. Depending on what your definition of 'effect' is, asking what effects can be applied is similar to asking for a list of everything that Pillow can do. If you would like such information, you can look at the documentation - https://pillow.readthedocs.io/en/stable/. You can read through the possible things to do with Pillow there.

**Comment by nissansz** [comment]
![image](https://github.com/python-pillow/Pillow/assets/41010669/a4b775db-d429-4343-bd79-f4bbd344668f)

**Reply by radarhere** [reply]
Please describe in words what you are after.

**Reply by nissansz** [reply]
I need effects for single lines like above

**Reply by radarhere** [reply]
It is still unclear what you would like.

If by 'effects for single lines', you are asking what possible transformations would could apply to a thin rectangle, this sounds like a previous question you asked - #7410. If you would like to know what it is possible to achieve with Pillow in a general sense, I encourage you to read the documentation.

**Reply by nissansz** [reply]
I am very interested in below two kinds of effects. Any method to imitate with pillow?

![image](https://github.com/python-pillow/Pillow/assets/41010669/62c51eb8-ef5d-473b-adeb-dc2700895b91)
![image](https://github.com/python-pillow/Pillow/assets/41010669/4c86e4e2-3150-45a5-90d7-1403712b0864)

I tried below filter and enhancers, but don't know how to achieve surprising effects.

```python
# 随机变换颜色
def random_color(image):
    enhancer = ImageEnhance.Color(image)
    factor = random.uniform(0.5, 1.8)  # 随机亮度因子
    return enhancer.enhance(factor)

# 随机变换亮度
def random_brightness(image):
    enhancer = ImageEnhance.Brightness(image)
    factor = random.uniform(0.5, 1.8)  # 随机亮度因子
    return enhancer.enhance(factor)

# 随机变换对比度
def random_contrast(image):
    enhancer = ImageEnhance.Contrast(image)
    factor = random.uniform(0.5, 1.8)  # 随机对比度因子
    return enhancer.enhance(factor)

# 随机变换对比度
def random_sharpness(image):
    enhancer = ImageEnhance.Sharpness(image)
    factor = random.uniform(0.4, 10)  # 随机对比度因子
    return enhancer.enhance(factor)

# 随机模糊
def random_blur(image):
    # blur_radius = random.random()  # 随机模糊半径
    blur_radius = random.uniform(0.1, 10)  # 随机对比度因子，越大越模糊
    print(blur_radius)
    return image.filter(ImageFilter.GaussianBlur(blur_radius))


def random_filter(image):
    r = random.randint(0,20)
    image1 = image
    if r ==0:
        return image1
    elif r == 1:
        image1 = image.filter(ImageFilter.CONTOUR)  # 白底黑线条
    elif r == 2:
        image1 = image.filter(ImageFilter.FIND_EDGES) # 黑底白线条
    elif r == 3:
        image1 = image.filter(ImageFilter.DETAIL)
    elif r == 4:
        image1 = image.filter(ImageFilter.EDGE_ENHANCE)
    elif r == 5:
        image1 = image.filter(ImageFilter.EDGE_ENHANCE_MORE)
    elif r == 6:
        image1 = image.filter(ImageFilter.EMBOSS)
    elif r == 7:
        image1 = image.filter(ImageFilter.SMOOTH)
    elif r == 8:
        image1 = image.filter(ImageFilter.SMOOTH_MORE)
    elif r == 9:
        image1 = image.filter(ImageFilter.SHARPEN)
    elif r == 10:
        image1 = image.filter(ImageFilter.Kernel((3, 3), (1, 1, 1, 0, 0, 0, 2, 0, 2)))
    # elif r == 11:
    #     image1 = image.filter(ImageFilter.RankFilter(5, 0))
    elif r == 12:
        image1 = image.filter(ImageFilter.RankFilter(5, random.randint(0, 24)))
    elif r == 13:
        image1 = image.filter(ImageFilter.MaxFilter(3))
    elif r == 14:
        image1 = image.filter(ImageFilter.MaxFilter(5))
    elif r == 15:
        image1 = image.filter(ImageFilter.MinFilter(3))
    elif r == 16:
        image1 = image.filter(ImageFilter.MinFilter(5))
    elif r == 17:
        image1 = image.filter(ImageFilter.MedianFilter(3))
    elif r == 18:
        image1 = image.filter(ImageFilter.MedianFilter(5))
    elif r == 19:
        image1 = image.filter(ImageFilter.ModeFilter(3))
    elif r == 20:
        image1 = image.filter(ImageFilter.ModeFilter(5))
    return image1
```

**Comment by nissansz** [comment]
Any method to simulate below effects?

![image](https://github.com/python-pillow/Pillow/assets/41010669/8ab07f12-be52-4288-a853-9065aebf1f8b)

**Reply by radarhere** [reply]
In one of your previous questions, you were able to add dots to text - https://github.com/python-pillow/Pillow/issues/7291#issuecomment-1639515989. That sounds like what you're asking for now, so you should be able to use the same method.

**Comment by radarhere** [comment]
Closing, as most of the questions here are single images, without any before version or description to explain what the requested transformation is.
